File: 1.two-sum.py
#
# @lc app=leetcode id=1 lang=python3
#
# [1] Two Sum
#

# @lc code=start
class Solution:
    def twoSum(self, nums: List[int], target: int) -> List[int]:
        seen = {}
        for i in range(len(nums)):
            # print(seen)
            if target - nums[i] in seen and seen[target-nums[i]] != i:
                    return [i,seen[target-nums[i]]]
            seen[nums[i]] = i


                
# @lc code=end

File: 1.two-sum.js
/*
 * @lc app=leetcode id=1 lang=javascript
 *
 * [1] Two Sum
 */

// @lc code=start
/**
 * @param {number[]} nums
 * @param {number} target
 * @return {number[]}
 */
var twoSum = function(nums, target) {
    let seen = {};
    for (let i=0;i<nums.length;i++){
        seek = target-nums[i]
        if (seen[seek] !== undefined){
            return [i,seen[seek]];
        }
        seen[nums[i]] = i;
    }
};
// @lc code=end

